src/Calculator.cpp

```diff
@@ -0,0 +1,106 @@
+            if (i == (equ_str.length() - 1)) {                  //If i is at the end of the vector,
+                equ_vec.push_back(num_str);                     //or the next character is not a number,
+            }                                                   //the current character is the last character in its number-string.
+            else if (char_is_num(equ_str[i + 1]) == false) {    //So, num_str is added to the end of the equ_vec vector
+                equ_vec.push_back(num_str);
+            }                                                   //If not, nothing is done.
+        }
+        else if (character == "+") {                            //If the current character is a +,
+            equ_vec.push_back(character);                       //add the string version to the end of equ_vec.
+        }
+        else {                                                  //If an unrecognized character is detected, print error message and abort loop
+            std::cout << "ERROR: Unrecognized input: " << character << std::endl;   // - needs implementation of actual error handling process
+            break;
+        }
+    }
+    return equ_vec;     //Return created vector
+}
+
+std::vector<std::string> add_sub_calc(std::vector<std::string> equ_vec) {
+    bool found_operator = false;            //Variable to track if we completed a pass through the equation without finding an operator
+    do {
+        found_operator = false;             //Reset found operator tracker to false
+        int result;                         //Variable to store the result
+        for (int i = 0; i < equ_vec.size(); i++) {   //Iterate over indices in equ_vec
+            if (equ_vec[i] == "+") {               //If the entry at current index is a +,
+                found_operator = true;      //Update operator tracker
+                result = std::stoi(equ_vec[(i - 1)]) + std::stoi(equ_vec[(i + 1)]);     //Convert the entries on either side of the operator to strings, and perform the operation
+                equ_vec.erase(equ_vec.begin() + (i - 1));           //Remove the left operand from the equation vector
+                equ_vec.erase(equ_vec.begin() + (i - 1));           //Remove the operator from the equation vector
+                equ_vec.erase(equ_vec.begin() + (i - 1));           //Remove the right operand from the equation vector
+                equ_vec.insert(equ_vec.begin() + (i - 1), std::to_string(result));  //Convert the result to a string, and insert it where the left operand was
+                break;      //End the for loop, so we can restart the iteration from the left side
+            }
+        }
+    } while (found_operator == true);   //If we found an operator, found_operator was updated to true, loop continues. 
+                                        //Otherwise, there are no more operators in the equation, found_operator remains false, loop ends
+    return equ_vec;     //Return vector with results of calculation

```

Comment:
if we modify the actual vector by reference, no need to return it later, so this function could be converted to void